### PR TITLE
[ENH] Implement s2s Linear Forecaster models family

### PR DESCRIPTION
**Pull request by rwtarpit**
#### Reference Issues/PRs
Implements LinearForecaster family of models (linear, nlinear and dlinear) following the PR :
[#3146](https://github.com/aeon-toolkit/aeon/pull/3146)

#### What does this implement/fix? Explain your changes.
This PR implements a simple 1 layered MLP based Linear Forecaster as a baseline for series2series deep learning forecasters.

#### Does your contribution introduce a new dependency? If yes, which one?
No

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [ ] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/forecasting/deep_learning/_ltsf_linear.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        import tensorflow as tf
+
+        rng = check_random_state(self.random_state)
+        self.random_state_ = rng.randint(0, np.iinfo(np.int32).max)
+
+        tf.keras.utils.set_random_seed(self.random_state_)
+
+        num_channels = input_shape[-1]
+        network = MLPNetwork(
+            n_layers=1, n_units=self.horizon * num_channels, activation="linear"
```
Comment (hadifawaz1999) [inline aeon/forecasting/deep_learning/_ltsf_linear.py]: both "dropout_rate" and "dropout_last" should be setup here, either to 0.0 if the original model does have dropout or to the correct values from the original model.

The "dropout_rate" is applied to every input Dense layer, and the "dropout_last" is applied at the output.

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
I took a first look, a small comment to begin,i will check the rest

**Review by hadifawaz1999 — DISMISSED** [review]
LGTM ! very nice, @MatthewMiddlehurst @TonyBagnall @alexbanwell1 let me know if you have any comments before merging

**Comment by alexbanwell1** [comment]
Looks good to me. Can't comment on the actual model but the forecasting module changes and series to series mixin look sensible

**Comment by MatthewMiddlehurst** [comment]
Also not familiar with the algorithm or whether this is a correct implementation, but would be good to discuss how to integrate series to series before putting this in probably. Thanks very much for the work though, this is helpful.

**Comment by rwtarpit** [comment]
I had a brief discussion on s2s and this linear forecaster with @hadifawaz1999 on discord. let me know whatever is decided and I refactor this accordingly.
thanks!

**Comment by TonyBagnall** [comment]
Hi @rwtarpit are you still working on this? I'll convert it to draft for now, feel free to reactivate if you want to finish this off
